html-server/src/main/kotlin/ru/testing/html/controller/Routing.kt

```diff
@@ -188,7 +193,7 @@ private suspend fun PipelineContext<Unit, ApplicationCall>.receiveTask(configura
             require(task != null) { "Task is not specified" }
             require(source != null) { "Source is not specified" }
             require(language != null) { "Language is not specified" }
-            val submission = makeSubmission(configuration, title, source, fileType = language, task)
+            val submission = makeSubmission(configuration, title, source, fileType = language, task, call.principal<UserPrincipal>()!!.user.id)
```

Comment:
Need if (call.pricipal<UserPricipal>() == null)?

Reply:
The same question